src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/DiagnosticTests.cs

```diff
@@ -47,4 +48,105 @@ public async Task Diagnostics_ShowErrors()
                 Assert.Equal("Counter.razor(7, 9): error CS0127: Since 'Counter.Function()' returns void, a return keyword must not be followed by an object expression", error);
             });
     }
+
+    [IdeFact]
+    public async Task Diagnostics_ShowErrors_Html()
+    {
+        // Arrange
+        await TestServices.SolutionExplorer.OpenFileAsync(RazorProjectConstants.BlazorProjectName, RazorProjectConstants.ErrorCshtmlFile, ControlledHangMitigatingCancellationToken);
+
+        await TestServices.Editor.SetTextAsync(@"
+@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+
+<!DOCTYPE html>
+<html lang=""en"">
+<head>
+</head>
+
+<body>
+    <p
+</body>
+</html>
+
+", ControlledHangMitigatingCancellationToken);
+
+        // Act
+        var errors = await TestServices.ErrorList.WaitForErrorsAsync("Error.cshtml", expectedCount: 1, ControlledHangMitigatingCancellationToken);
+
+        // Assert
+        Assert.Collection(errors,
+            (error) =>
+            {
+                Assert.Equal("Error.cshtml(10, 6): warning HTML0001: Element start tag is missing closing angle bracket.", error);
+            });
+    }
+
+    [IdeFact]
+    public async Task Diagnostics_ShowErrors_CSharp()
```

Comment:
Isn't this covered by `Diagnostics_ShowErrors_CSharpAndHtml` now?

Reply:
Yes, but I thought it was a bit rude to remove your test. Plus, I'm fine with more tests to be extra sure. What if my code fails when only one server returns diagnostics?